lib/modules/us_core_guidance/clinicalnotes_sequence.rb

```diff
@@ -51,175 +41,168 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+        document_attachments[patient_id] = attachments unless attachments.empty?
 
-        test_clinical_notes_document_reference('http://loinc.org|11488-4')
+        type_found
       end
 
-      test :have_discharge_summary do
-        metadata do
-          id '02'
-          name 'Server shall have Discharge Summary from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+      def check_diagnostic_report_required_category(patient_id)
+        resource_class = :DiagnosticReport.to_s
+        all_status = 'registered,partial,preliminary,final,amended,corrected,appended,cancelled,entered-in-error,unknown'
+        search_params = { 'patient': patient_id }
 
-        test_clinical_notes_document_reference('http://loinc.org|18842-5')
-      end
-
-      test :have_history_note do
-        metadata do
-          id '03'
-          name 'Server shall have History and Physical Note from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+        resources = reply_with_status_search(search_params, resource_class, all_status)
 
-        test_clinical_notes_document_reference('http://loinc.org|34117-2')
+        parse_diagnostic_report_resources(resources, resource_class, patient_id)
       end
 
-      test :have_procedures_note do
-        metadata do
-          id '04'
-          name 'Server returns Procedures Note from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+      def parse_diagnostic_report_resources(resources, resource_class, patient_id)
+        category_found = []
+        attachments = {}
 
-        test_clinical_notes_document_reference('http://loinc.org|28570-0')
-      end
+        resources&.select { |r| r.resourceType == resource_class }&.each do |resource|
+          resource.category&.each do |category|
+            code = category.coding.map { |coding| coding.code if CATEGORY_REQUIRED.include?(coding.code) }.compact
 
-      test :have_progress_note do
-        metadata do
-          id '05'
-          name 'Server returns Progress Note from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+            unless code.empty?
+              category_found << code.first unless category_found.include?(code.first)
```

Comment:
Why only include the first code?

Reply:
Line 156 saves the category code which matches clinical notes required category codes. Each DocumentReference instance has no more than one of such category code. Even if there are duplicated category code in the CodeableConcept, `.compact()` call removes such duplicates from result. So the `code`  array has zero or one category code.

Reply:
This is in the DiagnosticReport code, and category is 1..* for DiagnosticReports. Also, `compact` doesn't remove duplicates, it only removes `nil` elements.

Reply:
There is no overlap in these three categories. Will there be a report having both Cardiology and Radiology? 
Though I don't think this could happen, it is better to be extra cautious.

Reply:
Yeah, it's an extensible ValueSet, so we can't be sure that only those three categories will be used.